app/Controller/DictionariesController.php

```diff
@@ -40,6 +40,39 @@ class DictionariesController extends StandardController {
       'description' => 'asc'
     )
   );
+
+  /**
+   * Perform any dependency checks required prior to a delete operation.
+   * - postcondition: Session flash message updated (HTML) or HTTP status returned (REST)
+   *
+   * @since  COmanage Registry v4.0.1
+   * @param  Array Current data
+   * @return boolean true if dependency checks succeed, false otherwise.
+   */
+
+  function checkDeleteDependencies($curdata) {
+    // We can't delete a dictionary if it is linked to an attribute enumerator
+
+    if(!empty($curdata["AttributeEnumeration"])) {
+      $deleted_status = Hash::extract($curdata, 'AttributeEnumeration.{n}.deleted');
+
+      // There are no attribute enumerators here. Proceed to delete
+      if(!in_array(false, $deleted_status)) {
+        return true;
+      }
+
+      // Return the error message
+      if($this->request->is('restful')) {
+        $this->Api->restResultHeader(403, _txt('er.dict.with.attr'));
+      } else {
+        $this->Flash->set(_txt('er.dict.with.attr'), array('key' => 'error'));
```

Comment:
Since you have the context already anyway, it might be nice to include the ID or description of the Attribute Enumeration in the error string.

`'Dictionary is in use by Attribute Enumeration X and can not be deleted'`

Reply:
I can link a dictionary to multiple Attribute Enumerations. For the case of COUs, that can have multiple children, we return a generic message. Isn't this a similar use case?

Reply:
Done